File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Java/src/test/java/simulacionBandaMusica/utils/sorteoDatosTest.java
package simulacionBandaMusica.utils;

import org.junit.jupiter.api.Test;
import simulacionBandaMusica.enums.TipoGuitarra;
import simulacionBandaMusica.enums.TipoPercusion;

import static org.junit.jupiter.api.Assertions.*;
import static simulacionBandaMusica.utils.sorteoDatos.*;

class sorteoDatosTest {

    @Test
    void randomNameTest() {
        String generateName = randomName();

        int contadorCorrecto = 0;
        String[] catalogoNombresCorrectos = {
                "Pedro", "Alexandra", "Angel", "José",
                "Elena", "Ricardo", "Domingo", "Patricia",
                "Maribel", "Mariana", "Ivan", "Luis", "Mario", "Jorge", "Adrián", "Marcos", "Lara",
                "Marta", "Maria", "Claudia", "Sandra", "Fernando", "Roberto", "Marisa"
        };
        for (int i = 0; i < catalogoNombresCorrectos.length; i++) {
            if (catalogoNombresCorrectos[i].contains(generateName)) {
                assertTrue(true);
            } else {
                contadorCorrecto++;
            }
        }
        if (contadorCorrecto == 24) {
            fail();
        }

        int contadorIncorrecto = 0;
        String[] catalogoNombresIncorrectos = {"123", "...", ",", "??", "hola", "no es nombre", " ", "ç"};
        for (int i = 0; i < catalogoNombresIncorrectos.length; i++) {
            if (!catalogoNombresIncorrectos[i].contains(generateName)) {
                assertFalse(false);
                contadorIncorrecto++;
            }
        }
        if (contadorIncorrecto != 8) {
            fail();
        }
    }

    @Test
    void randomAnnoTest() {
        int annoTest = randomAnno();
        if (annoTest < 1 || annoTest > 35) {
            fail();
        }
    }

    @Test
    void randomPulmonarCapacityTest() {
        int pulomnarTest = randomPulmonarCapacity();
        if (pulomnarTest < 1 || pulomnarTest > 10) {
            fail();
        }
    }

    @Test
    void randomTipoGuitarraTest() {
        TipoGuitarra tipoTest = randomTipoGuitarra();
        if (!tipoTest.equals(TipoGuitarra.ACUSTICA) && !tipoTest.equals(TipoGuitarra.ELECTRICA)) {
            fail();
        }
    }

    @Test
    void randomNumCuerdasTest() {
        int numCuerdas = randomNumCuerdas();
        if (numCuerdas < 1 || numCuerdas > 6) {
            fail();
        }
    }

    @Test
    void randomNumTeclasTest() {
        int numTeclas = randomNumTeclas();
        if (numTeclas < 35 || numTeclas > 60) {
            fail();
        }
    }

    @Test
    void randomTipoPercusionTest() {
        TipoPercusion tipoTest = randomTipoPercusion();
        if (!tipoTest.equals(TipoPercusion.TAMBOR)
                && !tipoTest.equals(TipoPercusion.TIMBAL)
                && !tipoTest.equals(TipoPercusion.XILOFONO)
                && !tipoTest.equals(TipoPercusion.CAMPANA)
                && !tipoTest.equals(TipoPercusion.CROTALOS)
                && !tipoTest.equals(TipoPercusion.CELESTA)
                && !tipoTest.equals(TipoPercusion.CAJON)
                && !tipoTest.equals(TipoPercusion.TRIANGULO)) {
            fail();
        }
    }
}

File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Kotlin/src/test/kotlin/simulacionBandaMusica/utils/sorteoDatosTest.kt
package simulacionBandaMusica.utils

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import simulacionBandaMusica.enums.TipoGuitarra
import simulacionBandaMusica.enums.TipoPercusion
import simulacionBandaMusica.utils.sorteoDatos.randomAnno
import simulacionBandaMusica.utils.sorteoDatos.randomName
import simulacionBandaMusica.utils.sorteoDatos.randomNumCuerdas
import simulacionBandaMusica.utils.sorteoDatos.randomNumTeclas
import simulacionBandaMusica.utils.sorteoDatos.randomPulmonarCapacity
import simulacionBandaMusica.utils.sorteoDatos.randomTipoGuitarra
import simulacionBandaMusica.utils.sorteoDatos.randomTipoPercusion

internal class sorteoDatosTest {
    @Test
    fun randomNameTest() {
        val generateName: String = randomName()
        var contadorCorrecto = 0
        val catalogoNombresCorrectos = arrayOf(
            "Pedro", "Alexandra", "Angel", "José",
            "Elena", "Ricardo", "Domingo", "Patricia",
            "Maribel", "Mariana", "Ivan", "Luis", "Mario", "Jorge", "Adrián", "Marcos", "Lara",
            "Marta", "Maria", "Claudia", "Sandra", "Fernando", "Roberto", "Marisa"
        )
        for (i in catalogoNombresCorrectos.indices) {
            if (catalogoNombresCorrectos[i].contains(generateName)) {
                Assertions.assertTrue(true)
            } else {
                contadorCorrecto++
            }
        }
        if (contadorCorrecto == 24) {
            Assertions.fail<Any>()
        }
        var contadorIncorrecto = 0
        val catalogoNombresIncorrectos = arrayOf("123", "...", ",", "??", "hola", "no es nombre", " ", "ç")
        for (i in catalogoNombresIncorrectos.indices) {
            if (!catalogoNombresIncorrectos[i].contains(generateName)) {
                Assertions.assertFalse(false)
                contadorIncorrecto++
            }
        }
        if (contadorIncorrecto != 8) {
            Assertions.fail<Any>()
        }
    }

    @Test
    fun randomAnnoTest() {
        val annoTest: Int = randomAnno()
        if (annoTest < 1 || annoTest > 35) {
            Assertions.fail<Any>()
        }
    }

    @Test
    fun randomPulmonarCapacityTest() {
        val pulomnarTest: Int = randomPulmonarCapacity()
        if (pulomnarTest < 1 || pulomnarTest > 10) {
            Assertions.fail<Any>()
        }
    }

    @Test
    fun randomTipoGuitarraTest() {
        val tipoTest: TipoGuitarra = randomTipoGuitarra()
        if (tipoTest != TipoGuitarra.ACUSTICA && tipoTest != TipoGuitarra.ELECTRICA) {
            Assertions.fail<Any>()
        }
    }

    @Test
    fun randomNumCuerdasTest() {
        val numCuerdas: Int = randomNumCuerdas()
        if (numCuerdas < 1 || numCuerdas > 6) {
            Assertions.fail<Any>()
        }
    }

    @Test
    fun randomNumTeclasTest() {
        val numTeclas: Int = randomNumTeclas()
        if (numTeclas < 35 || numTeclas > 60) {
            Assertions.fail<Any>()
        }
    }

    @Test
    fun randomTipoPercusionTest() {
        val tipoTest: TipoPercusion = randomTipoPercusion()
        if (tipoTest != TipoPercusion.TAMBOR
            && tipoTest != TipoPercusion.TIMBAL
            && tipoTest != TipoPercusion.XILOFONO
            && tipoTest != TipoPercusion.CAMPANA
            && tipoTest != TipoPercusion.CROTALOS
            && tipoTest != TipoPercusion.CELESTA
            && tipoTest != TipoPercusion.CAJON
            && tipoTest != TipoPercusion.TRIANGULO
        ) {
            Assertions.fail<Any>()
        }
    }
}